src/mlpack/methods/ann/layer/softshrink.hpp

```diff
@@ -0,0 +1,224 @@
+ * http://www.opensource.org/licenses/BSD-3-Clause for more information.
+ */
+#ifndef MLPACK_METHODS_ANN_LAYER_SOFTSHRINK_HPP
+#define MLPACK_METHODS_ANN_LAYER_SOFTSHRINK_HPP
+
+#include <mlpack/prereqs.hpp>
+
+namespace mlpack {
+namespace ann /** Artifical Neural Network. */ {
+
+/**
+ * SoftShrinkage operator is defined as,
+ *         | x - lambda, if x >  lambda
+ *  f(x) = | x + lambda, if x < -lambda
+ *         | 0, otherwise
+ *
+ * lambda is set to 0.5 by default.
+ *
+ *          | 1 , if x >  lambda
+ *  f'(x) = | 1 , if x < -lambda
+ *          | 0 , otherwise
+ */
+template <
+    typename InputDataType = arma::mat,
+    typename OutputDataType = arma::mat
+>
+class SoftShrink
+{
+ public:
+  /**
+   * Create SoftShrink object using specified hyperparameter lambda.
+   * Hyperparameter lambda.
+   * Soft shrinkage has a threshold lambda which is calculated by multiplying the
+   * noise level sigma of the input(noisy image) and a coefficient 'a' which is one
+   * of the training parameter.
+   * Default (lambda = 0.5). lambda should be > 0.
+   * 
+   * @param lambda parameter
+   */
+  SoftShrink(const double lambda = 0.5);
+
+  /**
+   * Ordinary feed forward pass of a neural network, evaluating the function
+   * f(x) by propagating the activity forward through f.
+   * 
+   * @param input Input data used for evaluating the SoftShrink function.
+   * @param output Resulting output activation
+   */
+  template<typename InputType, typename OutputType>
+  void Forward(const InputType&& input, OutputType&& output);
+
+  /**
+   * Ordinary feed backward pass of a neural network, calculating the function
+   * f(x) by propagating x backwards through f. Using the results from the feed
+   * forward pass.
+   * 
+   * @param input The propagated input activation f(x).
+   * @param gy The backpropagated error.
+   * @param g The calculated gradient
+   */
+  template<typename DataType>
+  void Backward(const DataType&& input,
+                DataType&& gy,
+                DataType&& g);
+
+  //! Get the output parameter.
+  OutputDataType const& OutputParameter() const { return outputParameter; }
+  //! Modify the output parameter.
+  OutputDataType& OutputParameter() { return outputParameter; }
+
+  //! Get the delta.
+  OutputDataType const& Delta() const { return delta; }
+  //! Modify the delta.
+  OutputDataType& Delta() { return delta; }
+
+  //! Get the hyperparameter lambda.
+  double const& Lambda() const { return lambda; }
+  //! Modify the hyperparameter lambda.
+  double& Lambda() { return lambda; }
+
+  /**
+   * Serialize the layer.
+   */
+  template<typename Archive>
+  void serialize(Archive& ar, const unsigned int /* version */);
+
+ private:
+  /**
+   * Computes the value of Soft Shrink activation function.
+   *
+   * @param x Input data.
+   * @return f(x) 
+   */
+  double Fn(const double x)
+  {
+    if (x > lambda)
+    {
+      return x - lambda;
+    }
+    else if (x < -1 * lambda)
```

Comment:
You can also just write `-lambda`.

Reply:
Ok making the change right away.
Thanks.